biz.aQute.bndlib/src/aQute/bnd/classfile/ClassFile.java

```diff
@@ -153,4 +166,28 @@ static ByteBuffer slice(DataInput in, int length) throws IOException {
 		in.readFully(array, 0, length);
 		return ByteBuffer.wrap(array, 0, length);
 	}
+
+	public byte[] toBytes() throws IOException {
+		ByteArrayOutputStream bout = new ByteArrayOutputStream();
```

Comment:
We have tried very hard to never use ByteArrayInputStream in Bnd due to performance issues with over synchronization. Instead we use ByteBufferInputStream. But in this case simply use ByteBufferDataOutput since you need a DataOutput anyway.